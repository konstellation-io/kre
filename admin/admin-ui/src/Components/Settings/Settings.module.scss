@import 'Styles/colors';
@import 'Styles/mixins';
@import 'Styles/borders';
@import 'Styles/shadows';

$color-bg: palette(base);

.container {
  @include font-button;
  position: relative;
  line-height: 40px;
  height: 40px;
  text-align: center;
  padding: 0 12px;
  cursor: pointer;
  transition: color ease 0.4s;
  color: font-color(light);
  z-index: 2;

  &:hover {
    color: font-color(light);

    .arrow {
      border-color: white;
    }
  }

  &:active {
    box-shadow: none;
  }

  &:after {
    @include arrow(white);

    position: absolute;
    width: 7px;
    height: 7px;
    transform: rotate(45deg);
    right: 14px;
    top: 14px;
    content: '';
    transition: transform ease 0.3s, top ease 0.3s;
  }

  &.is-open:after {
    transform: rotate(45deg) scale(-1);
    top: 17px;
  }
}

.label {
  display: flex;
  flex-direction: row;
  padding: 0 20px;
  user-select: none;
}

.options {
  @include shadow(4);

  transition: max-height ease 0.3s;
  overflow: hidden;
  background-color: $color-bg;
}
