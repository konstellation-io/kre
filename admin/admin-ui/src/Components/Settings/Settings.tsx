import { BUTTON_ALIGN, Button, useClickOutside } from 'kwc';
import React, { memo, useEffect, useRef, useState } from 'react';

import AuditIcon from '@material-ui/icons/EventNote';
import { ENDPOINT } from 'Constants/application';
import LogoutIcon from '@material-ui/icons/ExitToApp';
import ProfileIcon from '@material-ui/icons/Person';
import ROUTE from 'Constants/routes';
import SettingsIcon from '@material-ui/icons/VerifiedUser';
import UsersIcon from '@material-ui/icons/SupervisorAccount';
import { checkPermission } from '../../rbac-rules';
import cx from 'classnames';
import { get } from 'lodash';
import styles from './Settings.module.scss';
import useEndpoint from 'Hooks/useEndpoint';
import { useHistory } from 'react-router';
import useLogin from 'Graphql/hooks/useLogin';
import useUserAccess from 'Hooks/useUserAccess';

const BUTTON_HEIGHT = 40;
const buttonStyle = {
  paddingLeft: '20%'
};

type Props = {
  label: string;
};

function Settings({ label }: Props) {
  const { logout } = useLogin();
  const { accessLevel } = useUserAccess();
  const history = useHistory();
  const [opened, setOpened] = useState(false);
  const [logoutResponse, doLogout] = useEndpoint({
    endpoint: ENDPOINT.LOGOUT,
    method: 'POST'
  });
  const settingsRef = useRef(null);
  const { addClickOutsideEvents, removeClickOutsideEvents } = useClickOutside({
    componentRef: settingsRef,
    action: close
  });

  useEffect(() => {
    if (logoutResponse.complete) {
      if (get(logoutResponse, 'status') === 200) {
        logout();
        history.push(ROUTE.LOGIN);
      } else {
        console.error(`Error sending logout request.`);
      }
    }
  }, [logoutResponse, history, logout]);

  function close() {
    setOpened(false);
    removeClickOutsideEvents();
  }

  function open() {
    if (!opened) {
      setOpened(true);
      addClickOutsideEvents();
    }
  }

  function getBaseProps(label: string) {
    return {
      label: label.toUpperCase(),
      key: `button${label}`,
      align: BUTTON_ALIGN.LEFT,
      style: buttonStyle
    };
  }

  const settingsButton = (
    <Button
      {...getBaseProps('Settings')}
      to={ROUTE.SETTINGS}
      Icon={SettingsIcon}
    />
  );
  const usersButton = (
    <Button {...getBaseProps('Users')} to={ROUTE.USERS} Icon={UsersIcon} />
  );
  const profileButton = (
    <Button
      {...getBaseProps('Profile')}
      to={ROUTE.PROFILE}
      Icon={ProfileIcon}
    />
  );
  const auditButton = (
    <Button {...getBaseProps('Audit')} to={ROUTE.AUDIT} Icon={AuditIcon} />
  );
  const logoutButton = (
    <Button
      {...getBaseProps('Logout')}
      onClick={() => doLogout()}
      Icon={LogoutIcon}
    />
  );

  const buttons: JSX.Element[] = [];
  if (checkPermission(accessLevel, 'settings:edit')) {
    buttons.push(settingsButton);
  }
  if (checkPermission(accessLevel, 'users:edit')) {
    buttons.push(usersButton);
  }
  if (checkPermission(accessLevel, 'audit:view')) {
    buttons.push(auditButton);
  }
  buttons.push(profileButton, logoutButton);

  const nButtons = buttons.length;
  const optionsHeight = nButtons * BUTTON_HEIGHT;

  return (
    <div
      className={cx(styles.container, { [styles['is-open']]: opened })}
      onClick={open}
      data-testid="settingsContainer"
    >
      <div className={styles.label} data-testid="settings-label">
        {label}
      </div>
      <div
        ref={settingsRef}
        className={styles.options}
        style={{ maxHeight: opened ? optionsHeight : 0 }}
        data-testid="settingsContent"
      >
        {buttons}
      </div>
    </div>
  );
}

export default memo(Settings);
