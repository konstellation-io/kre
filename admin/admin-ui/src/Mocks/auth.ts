import { AccessLevel } from 'Graphql/types/globalTypes';
import { loader } from 'graphql.macro';

const GetUserEmailQuery = loader('Graphql/queries/getMe.graphql');

export const usernameMock = {
  request: {
    query: GetUserEmailQuery
  },
  result: {
    data: {
      me: {
        email: 'user@konstellation.com',
        id: 'someId',
        accessLevel: AccessLevel.ADMIN
      }
    }
  }
};

export const unauthorizedUsernameMock = {
  request: {
    query: GetUserEmailQuery
  },
  result: {
    data: {
      me: {
        email: 'unauthorizedUser@konstellation.com',
        id: 'someId',
        accessLevel: AccessLevel.ADMIN
      }
    }
  },
  error: new Error('Unauthorized user')
};
