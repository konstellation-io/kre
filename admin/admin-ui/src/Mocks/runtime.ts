import { runtime, version } from './version';

import { loader } from 'graphql.macro';

const GetRuntimesQuery = loader('Graphql/queries/getRuntimes.graphql');
const GetRuntimeAndVersionQuery = loader(
  'Graphql/queries/getRuntimeAndVersions.graphql'
);
const CreateRuntimeMutation = loader('Graphql/mutations/createRuntime.graphql');
const RuntimeCreatedSubscription = loader(
  'Graphql/subscriptions/runtimeCreated.graphql'
);

export const dashboardMock = {
  request: {
    query: GetRuntimesQuery
  },
  result: {
    data: {
      runtimes: [
        {
          id: '00001',
          name: 'Some Name',
          description: 'Some Description',
          status: 'STARTED',
          creationDate: '2019-11-28T15:28:01+00:00',
          creationAuthor: { email: 'some@user.com' },
          publishedVersion: {
            status: 'STARTED'
          }
        },
        {
          id: '00002',
          name: 'Some Other Name',
          description: 'Some Other Description',
          status: 'STARTED',
          creationDate: '2019-11-27T15:28:01+00:00',
          creationAuthor: { email: 'some@user.com' },
          publishedVersion: {
            status: 'STARTED'
          }
        },
        {
          id: '00003',
          name: 'Creating runtime',
          description: 'Some Description',
          status: 'CREATING',
          creationDate: '2019-11-27T15:28:01+00:00',
          creationAuthor: { email: 'some@user.com' },
          publishedVersion: {
            status: 'CREATING'
          }
        }
      ]
    }
  }
};

export const dashboardErrorMock = {
  request: {
    query: GetRuntimesQuery
  },
  error: new Error('cannot get runtimes')
};

export const addRuntimeMock = {
  request: {
    query: CreateRuntimeMutation,
    variables: { input: { name: 'New Runtime' } }
  },
  result: {
    data: {
      createRuntime: { name: 'some name' }
    }
  }
};

export const getRuntimeAndVersionsMock = {
  request: {
    query: GetRuntimeAndVersionQuery
  },
  result: {
    data: {
      runtime,
      version
    }
  }
};

export const getRuntimeAndVersionsErrorMock = {
  request: {
    query: GetRuntimeAndVersionQuery
  },
  error: new Error('cannot get runtime and versions')
};

export const runtimeCreatedMock = {
  request: {
    query: RuntimeCreatedSubscription
  },
  result: {
    data: {
      runtimeCreated: { id: 'some id', name: 'some name' }
    }
  }
};
