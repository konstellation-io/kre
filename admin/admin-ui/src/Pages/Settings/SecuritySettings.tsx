import React from 'react';
import SettingsHeader from './components/SettingsHeader/SettingsHeader';
import AllowedDomains from './components/AllowedDomains/AllowedDomains';

import cx from 'classnames';
import styles from './Settings.module.scss';

function SecuritySettings() {
  return (
    <>
      <div className={cx(styles.form, styles.securitySettings)}>
        <SettingsHeader>Security settings</SettingsHeader>
        <AllowedDomains />
      </div>
    </>
  );
}

export default SecuritySettings;
