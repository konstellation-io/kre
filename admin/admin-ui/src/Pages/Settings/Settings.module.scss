@import 'Styles/colors';
@import 'Styles/mixins';

$color-bg: palette(base, 800);
$color-bg-header: palette(base, 600);

.container {
  height: 100%;
  display: flex;
  flex-direction: row;
  width: 100%;
  background-color: $color-bg;
}

.content {
  height: 100%;
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  overflow: auto;
}

.form {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  min-width: 540px;
  overflow: hidden;
}
.generalSettings {
  .formField {
    height: 140px;
    display: flex;
    flex-direction: row;
    align-items: center;

    .label {
      @include font-body;
      color: font-color(light);

      margin: 0 40px 0 30px;
    }

    svg {
      margin-left: 20px;
    }

    .input {
      width: 140px;
    }
  }
}

.securitySettings {
  .tabs {
    background-color: $color-bg-header;
  }
  background-color: $color-bg;
}

.header {
  padding: 3 * $grid-unit;
  background-color: $color-bg-header;
}

.formTitle {
  @include font-headline;
  color: font-color(lowlight);
}

.formDescription {
  @include font-body;

  margin-top: 10px;
}

.listsContainer {
  display: flex;
  overflow: hidden;
  height: 100vh;
  & > div:first-child {
    margin-right: 48px;
  }
}

@media (max-width: 1400px) {
  .listsContainer {
    flex-direction: column;
    & div:first-child {
      margin-right: 0;
    }
  }
}
