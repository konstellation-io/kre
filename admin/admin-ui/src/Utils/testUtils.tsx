import React, { ReactElement } from 'react';
import { createMemoryHistory } from 'history';
import { Router, Route } from 'react-router-dom';
import { MemoryRouter } from 'react-router';
import { render, RenderResult } from '@testing-library/react';
import { mount } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';

type TestHookprops = {
  callback: Function;
};
const TestHook = ({ callback }: TestHookprops) => {
  callback();
  return null;
};

export const testHook = (callback: Function) => {
  mount(<TestHook callback={callback} />);
};

interface RenderWithRouterEl extends RenderResult {
  customRerender: Function;
}

export function renderWithRouter(
  component: ReactElement,
  route: string = '/path/:someParam/index',
  param = { param: 'someParam', value: 'someValue' }
) {
  const history = createMemoryHistory();
  history.push(route.replace(param.param, param.value));

  const element = render(
    <MemoryRouter>
      <Router history={history}>
        <Route path={route}>{component}</Route>
      </Router>
    </MemoryRouter>
  ) as RenderWithRouterEl;

  element.customRerender = function(newComponent: ReactElement) {
    element.rerender(<MemoryRouter>{newComponent}</MemoryRouter>);
  };

  return {
    element,
    history
  };
}
