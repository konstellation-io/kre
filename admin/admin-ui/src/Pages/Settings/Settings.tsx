import GeneralIcon from '@material-ui/icons/DeviceHub';
import GeneralSettings from './GeneralSettings';
import PageBase from 'Components/Layout/PageBase/PageBase';
import ROUTE from 'Constants/routes';
import React from 'react';
import { Route } from 'react-router';
import SecurityIcon from '@material-ui/icons/Security';
import SecuritySettings from './SecuritySettings';
import Sidebar from 'Components/Sidebar/Sidebar';
import UsersIcon from '@material-ui/icons/SupervisorAccount';
import UsersSettings from './pages/UsersSettings/UsersSettings';
import styles from './Settings.module.scss';

const tabs = [
  {
    label: 'GENERAL',
    route: ROUTE.SETTINGS_GENERAL,
    Icon: GeneralIcon
  },
  {
    label: 'SECURITY',
    route: ROUTE.SETTINGS_SECURITY,
    Icon: SecurityIcon
  },
  {
    label: 'USERS',
    route: ROUTE.SETTINGS_USERS,
    Icon: UsersIcon
  }
];

function Settings() {
  return (
    <PageBase>
      <div className={styles.container} data-testid="settingsContainer">
        <Sidebar title="Settings" tabs={tabs} />
        <div className={styles.content}>
          <Route
            exact
            path={ROUTE.SETTINGS_GENERAL}
            component={GeneralSettings}
          />
          <Route
            exact
            path={ROUTE.SETTINGS_SECURITY}
            component={SecuritySettings}
          />
          <Route exact path={ROUTE.SETTINGS_USERS} component={UsersSettings} />
        </div>
      </div>
    </PageBase>
  );
}

export default Settings;
