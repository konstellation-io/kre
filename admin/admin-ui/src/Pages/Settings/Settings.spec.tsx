import * as MOCK from 'Mocks/settings';

import { History, createMemoryHistory } from 'history';
import { RenderResult, cleanup, fireEvent } from '@testing-library/react';

import { MockedProvider } from '@apollo/client/testing';
import ROUTE from 'Constants/routes';
import React from 'react';
import { Router } from 'react-router-dom';
import Settings from './Settings';
import { act } from 'react-dom/test-utils';
import { dashboardMock } from 'Mocks/runtime';
import { renderWithRouter } from 'Utils/testUtils';
import { usernameMock } from 'Mocks/auth';
import wait from 'waait';

const mocks = [
  MOCK.expirationTimeMock,
  MOCK.domainMock,
  MOCK.usersActivityMock,
  MOCK.addAllowedDomainMock,
  MOCK.updateExpirationTime,
  dashboardMock,
  usernameMock
];

afterEach(cleanup);

function generateComponent() {
  const history = createMemoryHistory();
  history.push(ROUTE.SETTINGS);

  const wrapper = renderWithRouter(
    <MockedProvider mocks={mocks} addTypename={false}>
      <Router history={history}>
        <Settings />
        {/* <Routes /> */}
      </Router>
    </MockedProvider>
  );

  return [wrapper.element, history];
}

it('Renders Settings without crashing', () => {
  const { container } = generateComponent()[0] as RenderResult;
  expect(container).toMatchSnapshot();
});

it('Shows general settings by default', async () => {
  const { getByText } = generateComponent()[0] as RenderResult;

  await act(async () => {
    await wait(0);
  });

  expect(getByText('GENERAL')).toBeInTheDocument();
});

it('can move to other settings', async () => {
  const { container, getByText } = generateComponent()[0] as RenderResult;

  await act(async () => {
    await wait(0);
  });

  fireEvent.click(container, getByText('SECURITY'));
  expect(getByText('SECURITY')).toBeInTheDocument();

  fireEvent.click(container, getByText('GENERAL'));
  expect(getByText('GENERAL')).toBeInTheDocument();
});

// FIXME: Research how to mock react-hook-form properly in order to trigger errors
it.skip('Security settings works properly', async () => {
  const [{ getByText, getByTestId }, history] = generateComponent() as [
    RenderResult,
    History
  ];

  history.push(ROUTE.SETTINGS_SECURITY);

  await act(async () => {
    await wait(0);
  });

  expect(getByText('Security settings')).toBeInTheDocument();
  expect(getByText('domain.1')).toBeInTheDocument();

  fireEvent.change(getByTestId('input'), { target: { value: 'intelygenz' } });
  fireEvent.click(getByText('ADD DOMAIN'));

  expect(getByTestId('error-message').textContent).not.toBe('');

  fireEvent.change(getByTestId('input'), {
    target: { value: 'intelygenz.com' }
  });
  fireEvent.click(getByText('ADD DOMAIN'));

  await act(async () => {
    await wait(0);
  });
  expect(getByText('intelygenz.com')).toBeInTheDocument();
});
