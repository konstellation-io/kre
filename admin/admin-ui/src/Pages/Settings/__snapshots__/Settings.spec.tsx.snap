// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Settings without crashing 1`] = `
<div>
  <div
    class="splash"
    data-testid="splashscreen"
  />
  <div
    class="content"
  >
    <div
      class="_3saxo"
      data-testid="spinner"
      style="width: 125px; height: 125px;"
    >
      <div
        style="width: 125px; height: 125px; overflow: hidden; margin: 0px auto; outline: none;"
      >
        <svg
          height="240"
          preserveAspectRatio="xMidYMid meet"
          style="width: 100%; height: 100%; transform: translate3d(0,0,0);"
          viewBox="0 0 240 240"
          width="240"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <clippath
              id="__lottie_element_2"
            >
              <rect
                height="240"
                width="240"
                x="0"
                y="0"
              />
            </clippath>
            <clippath
              id="__lottie_element_5"
            >
              <path
                d="M0,0 L240,0 L240,240 L0,240z"
              />
            </clippath>
            <clippath
              id="__lottie_element_15"
            >
              <path
                d="M0,0 L240,0 L240,240 L0,240z"
              />
            </clippath>
            <clippath
              id="__lottie_element_25"
            >
              <path
                d="M0,0 L240,0 L240,240 L0,240z"
              />
            </clippath>
            <clippath
              id="__lottie_element_35"
            >
              <path
                d="M0,0 L240,0 L240,240 L0,240z"
              />
            </clippath>
            <clippath
              id="__lottie_element_45"
            >
              <path
                d="M0,0 L240,0 L240,240 L0,240z"
              />
            </clippath>
            <clippath
              id="__lottie_element_55"
            >
              <path
                d="M0,0 L240,0 L240,240 L0,240z"
              />
            </clippath>
            <clippath
              id="__lottie_element_65"
            >
              <path
                d="M0,0 L240,0 L240,240 L0,240z"
              />
            </clippath>
            <clippath
              id="__lottie_element_75"
            >
              <path
                d="M0,0 L240,0 L240,240 L0,240z"
              />
            </clippath>
          </defs>
          <g
            clip-path="url(#__lottie_element_2)"
          >
            <g
              opacity="1"
              style="display: block;"
              transform="matrix(0.8999999761581421,0,0,0.8999999761581421,120,120)"
            >
              <g
                opacity="1"
                transform="matrix(1,0,0,1,0,0)"
              >
                <path
                  d=" M0,-65 C35.87350082397461,-65 65,-35.87350082397461 65,0 C65,35.87350082397461 35.87350082397461,65 0,65 C-35.87350082397461,65 -65,35.87350082397461 -65,0 C-65,-35.87350082397461 -35.87350082397461,-65 0,-65z"
                  fill-opacity="0"
                  stroke="rgb(0,207,255)"
                  stroke-linecap="butt"
                  stroke-linejoin="miter"
                  stroke-miterlimit="4"
                  stroke-opacity="1"
                  stroke-width="2"
                />
              </g>
            </g>
            <g
              opacity="0.19"
              style="display: block;"
              transform="matrix(-0.12860608100891113,-0.09169714897871017,0.09169714897871017,-0.12860608100891113,175.84800720214844,108.85103607177734)"
            >
              <g
                opacity="1"
                transform="matrix(1,0,0,1,0,0)"
              >
                <path
                  d=" M0,-120 C66.22799682617188,-120 120,-66.22799682617188 120,0 C120,66.22799682617188 66.22799682617188,120 0,120 C-66.22799682617188,120 -120,66.22799682617188 -120,0 C-120,-66.22799682617188 -66.22799682617188,-120 0,-120z"
                  fill="rgb(0,207,255)"
                  fill-opacity="1"
                />
              </g>
            </g>
            <g
              opacity="0.9500000000000001"
              style="display: block;"
              transform="matrix(-0.08120838552713394,-0.057902220636606216,0.057902220636606216,-0.08120838552713394,175.84800720214844,108.85103607177734)"
            >
              <g
                opacity="1"
                transform="matrix(1,0,0,1,0,0)"
              >
                <path
                  d=" M0,-120 C66.22799682617188,-120 120,-66.22799682617188 120,0 C120,66.22799682617188 66.22799682617188,120 0,120 C-66.22799682617188,120 -120,66.22799682617188 -120,0 C-120,-66.22799682617188 -66.22799682617188,-120 0,-120z"
                  fill="rgb(0,207,255)"
                  fill-opacity="1"
                />
              </g>
            </g>
            <g
              clip-path="url(#__lottie_element_75)"
              opacity="1"
              style="display: block;"
              transform="matrix(0.5224999785423279,0.000047123889089562,-0.000047123889089562,0.5224999785423279,101.39936065673828,18.396812438964844)"
            >
              <g
                opacity="0.2"
                style="display: block;"
                transform="matrix(0,0,0,0,120,120)"
              >
                <g
                  opacity="1"
                  transform="matrix(1,0,0,1,0,0)"
                >
                  <path
                    d=" M0,-120 C66.22799682617188,-120 120,-66.22799682617188 120,0 C120,66.22799682617188 66.22799682617188,120 0,120 C-66.22799682617188,120 -120,66.22799682617188 -120,0 C-120,-66.22799682617188 -66.22799682617188,-120 0,-120z"
                    fill="rgb(0,207,255)"
                    fill-opacity="1"
                  />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
            </g>
            <g
              clip-path="url(#__lottie_element_65)"
              style="display: none;"
            >
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
            </g>
            <g
              clip-path="url(#__lottie_element_55)"
              style="display: none;"
            >
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
            </g>
            <g
              clip-path="url(#__lottie_element_45)"
              style="display: none;"
            >
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
            </g>
            <g
              clip-path="url(#__lottie_element_35)"
              style="display: none;"
            >
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
            </g>
            <g
              clip-path="url(#__lottie_element_25)"
              style="display: none;"
            >
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
            </g>
            <g
              clip-path="url(#__lottie_element_15)"
              style="display: none;"
            >
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
            </g>
            <g
              clip-path="url(#__lottie_element_5)"
              style="display: none;"
            >
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
    </div>
    <div
      class="container"
      data-testid="settingsContainer"
    >
      <div
        class="container"
      >
        <div
          class="header"
        >
          <div
            class="back"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root icon-regular"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
              />
            </svg>
            <span
              class="title"
            >
              Settings
            </span>
          </div>
          <div
            class="description"
          />
        </div>
        <div
          class="navButtons"
        >
          <div>
            <a
              class=""
              href="/settings/general"
            >
              <div
                class="item size_m"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root icon-regular"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 16l-4-4V8.82C14.16 8.4 15 7.3 15 6c0-1.66-1.34-3-3-3S9 4.34 9 6c0 1.3.84 2.4 2 2.82V12l-4 4H3v5h5v-3.05l4-4.2 4 4.2V21h5v-5h-4z"
                  />
                </svg>
                <span>
                  GENERAL
                </span>
              </div>
            </a>
            <a
              class=""
              href="/settings/security"
            >
              <div
                class="item size_m"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root icon-regular"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11v8.8z"
                  />
                </svg>
                <span>
                  SECURITY
                </span>
              </div>
            </a>
            <a
              class=""
              href="/settings/users"
            >
              <div
                class="item size_m"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root icon-regular"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.5 12c1.38 0 2.49-1.12 2.49-2.5S17.88 7 16.5 7C15.12 7 14 8.12 14 9.5s1.12 2.5 2.5 2.5zM9 11c1.66 0 2.99-1.34 2.99-3S10.66 5 9 5C7.34 5 6 6.34 6 8s1.34 3 3 3zm7.5 3c-1.83 0-5.5.92-5.5 2.75V19h11v-2.25c0-1.83-3.67-2.75-5.5-2.75zM9 13c-2.33 0-7 1.17-7 3.5V19h7v-2.25c0-.85.33-2.34 2.37-3.47C10.5 13.1 9.66 13 9 13z"
                  />
                </svg>
                <span>
                  USERS
                </span>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="content"
      />
    </div>
  </div>
</div>
`;
